test/integration/transactions/transactions.spec.test.js

```diff
@@ -119,3 +135,79 @@ describe('Transactions Spec Legacy Tests', function () {
 
   generateTopologyTests(testSuites, testContext, testFilter);
 });
+
+describe('Transactions', function () {
+  const dbName = 'retryable-handshake-tests';
+  const collName = 'coll';
+  const docs = [{ _id: 1, x: 11 }];
+  let client;
+  let db;
+  let coll;
+
+  beforeEach(function () {
+    if (
```

Comment:
should there be an auth requirement and a skip on serverless requirement in addition to the ones below?

Reply:
Added auth and serverless checks.

Reply:
I think we should be able to include the runOn checks for server version, topology, and auth in the metadata on the `it` blocks (similar to how we do in `enumerate_databases.test.ts` for example), which would allow our runner to automatically attach the correct skip reason, then we can update the check below to only check for SERVERLESS explicitly and update the skipReason text accordingly

Reply:
updated this